lib/superintendent/request/params_case_translator.rb

```diff
@@ -27,10 +27,31 @@ def underscored_keys(value)
       when Array
         value.map { |v| underscored_keys(v) }
       when Hash
+        if value['data'].present?
+          transform_type_to_snake_case(value['data'])
+        end
```

Comment:
Good call. We should add a test similar to the first one where we don't expect type to change with something like:

 { 'data' => { 'type' => 'dataElements', 'settings' => {'type' => 'Foo'} } }

Then we can still do something similar to what I suggested.
```
if value.dig('data', 'type').present?
  value['data']['type'] = underscored_key(value['data']['type'])
end
```